android-sdk/src/main/java/co/elastic/apm/android/sdk/traces/session/impl/DefaultSessionIdProvider.java

```diff
@@ -0,0 +1,67 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *	http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package co.elastic.apm.android.sdk.traces.session.impl;
+
+import java.util.UUID;
+import java.util.concurrent.TimeUnit;
+
+import co.elastic.apm.android.sdk.traces.session.SessionIdProvider;
+
+/**
+ * Provides an in-memory id that has a 30 mins timeout that gets reset on every call to
+ * {@link SessionIdProvider#getSessionId()} - If 30 mins or more have passed since the last call,
+ * then a new session id is generated.
+ */
+public class DefaultSessionIdProvider implements SessionIdProvider {
+    private final CurrentTimeMillisProvider currentTimeMillisProvider;
+    private long expireTimeMillis;
+    private String sessionId;
+
+    DefaultSessionIdProvider(CurrentTimeMillisProvider currentTimeMillisProvider) {
+        this.currentTimeMillisProvider = currentTimeMillisProvider;
+    }
+
+    public DefaultSessionIdProvider() {
+        this(new SystemCurrentTimeMillisProvider());
+    }
+
+    @Override
+    public String getSessionId() {
+        verifySessionExpiration();
+        if (sessionId == null) {
+            sessionId = generateSessionId();
+        }
+        scheduleExpireTime();
```

Comment:
At first, I was a little confused why we re-schedule on every access but I get it now - we want to generate a new ID after 30min of inactivity. Makes sense.

Reply:
Yes it's correct, this default behaviour is about having 30 mins as an idle timeout